File: software/lambda-js/src/util/Log.ts
enum LogLevel {
    DEBUG,
    INFO,
    ERROR 
}
type LogLevelStrings = keyof typeof LogLevel;

const LOG_LEVEL_STRING: LogLevelStrings = <LogLevelStrings> process.env.LOG_LEVEL ?? "INFO";
const LOG_LEVEL: LogLevel = LogLevel[LOG_LEVEL_STRING];

export default {
    shouldLog(inputLevel: LogLevel): boolean {
        return inputLevel >= LOG_LEVEL;
    },
    debug(message?: any) {
        if (this.shouldLog(LogLevel.DEBUG)) { console.debug(message) }
    },
    info(message?: any) {
        if (this.shouldLog(LogLevel.INFO)) { console.info(message) }
    },
    error(message?: any) {
        if (this.shouldLog(LogLevel.ERROR)) { console.error(message) }
    }
}

File: software/lambda/common/src/main/kotlin/io/mycdkapp/common/debug/Log.kt
package io.mycdkapp.common.debug

object Log {
    private val logLevel by lazy {
        if (System.getenv("LOG_LEVEL").isNullOrBlank())
            LogLevel.INFO
        else
            LogLevel.valueOf(System.getenv("LOG_LEVEL"))
    }

    enum class LogLevel(val value: Int) {
        DEBUG(1),
        INFO(2),
        ERROR(3)
    }

    private fun log(level: LogLevel, message: String) {
        if (logLevel.value <= level.value) println("[${level.name}] $message")
    }

    fun debug(message: String) {
        log(LogLevel.DEBUG, message)
    }

    fun info(message: String) {
        log(LogLevel.INFO, message)
    }

    fun error(message: String) {
        log(LogLevel.ERROR, message)
    }
}